sycl/include/CL/sycl/detail/kernel_desc.hpp

```diff
@@ -75,6 +77,13 @@ template <class KernelNameType> struct KernelInfo {
   }
   static constexpr const char *getName() { return ""; }
   static constexpr bool isESIMD() { return 0; }
+#ifndef DNDEBUG
+  static constexpr const char *getFileName() { return ""; }
+  static constexpr const char *getFunctionName() { return ""; }
+  static constexpr unsigned getLineNumber() { return 0; }
+  static constexpr unsigned getColumnNumber() { return 0; }
+#endif
```

Comment:
I think we would prefer it the other way around: methods always exist in the class, but they only return meaningful data if NDEBUG is not defined. Also, there's a small typo:
```suggestion
#ifndef NDEBUG
  static constexpr const char *getFileName() { return ""; }
  static constexpr const char *getFunctionName() { return ""; }
  static constexpr unsigned getLineNumber() { return 0; }
  static constexpr unsigned getColumnNumber() { return 0; }
#endif
```

Reply:
@elizabethandrews - here